__tests__/infrastructure/TransactionService.spec.js

```diff
@@ -70,53 +64,174 @@ describe('Transaction Service', () => {
-				...pageInfo,
-				data: [
-					TestHelper.mockTransaction(mockBlockInfo)
-				]
+			const partialTransactionFromSDK = {
+				...partialTransaction,
+				deadline: 1665914320,
+				maxFee: '0.050000',
+				type: 16961,
+				innerTransactions: [
+					{
+						...innerTransaction,
+						deadline: '2022-02-28 17:55:31',
+						maxFee: 0,
+						transactionType: 16724,
+						transactionBody: {
+							transactionType: 16724,
+							message: {
+								type: -1,
+								payload: ''
+							},
+							recipient: 'TDCPHIHQPN6WKJJIUCOFISJCB4NULEZOS4NCQQQ',
+							mosaics: [
+								{
+									amount: '5',
+									mosaicId: '7F2D26E89342D398',
+									mosaicAliasName: [
+										'teria.revokable'
+									]
+								}
+							]
+						}
+					}
+				],
+				transactionInfo: {
+					height: 0,
+					index: 0,
+					timestamp: 1637848847,
+					feeMultiplier: 0,
+					hash: 'C3A9E202E14EA5890914F9884627A280CDD7571C0E3EEE0D6FF7788626136D4B',
+					merkleComponentHash: '0000000000000000000000000000000000000000000000000000000000000000'
+				},
+				transactionBody: {
+					transactionType: 16961
+				}
 			};
 
-			const searchTransactions = stub(TransactionService, 'searchTransactions');
-			searchTransactions.returns(Promise.resolve(mockSearchTransactions));
+			stub(TransactionService, 'getTransactionStatus').returns(Promise.resolve(transactionStatus));
+
+			stub(TransactionService, 'getTransaction').returns(Promise.resolve(partialTransaction));
+
+			stub(TransactionService, 'createTransactionFromSDK').returns(Promise.resolve(partialTransactionFromSDK));
 
 			// Act:
-			const transactionList = await TransactionService.getTransactionList(pageInfo, {});
+			const {
+				blockHeight,
+				transactionBody,
+				transactionHash,
+				maxFee,
+				timestamp,
+				status,
+				confirm
+			} = await TransactionService.getTransactionInfo('C3A9E202E14EA5890914F9884627A280CDD7571C0E3EEE0D6FF7788626136D4B');
 
 			// Assert:
-			expect(transactionList.totalRecords).toEqual(500);
-			expect(transactionList.pageNumber).toEqual(pageInfo.pageNumber);
-			expect(transactionList.pageSize).toEqual(pageInfo.pageSize);
-			expect(transactionList.data).toHaveLength(1);
-			transactionList.data.forEach(transaction => {
-				expect(transaction).toHaveProperty('transactionHash');
-				expect(transaction).toHaveProperty('transactionType');
-				expect(transaction).toHaveProperty('recipient');
-				expect(transaction).toHaveProperty('extendGraphicValue');
+			expect(blockHeight).toEqual(undefined);
+			expect(transactionHash).toEqual(partialTransactionFromSDK.transactionInfo.hash);
+			expect(maxFee).toEqual(partialTransactionFromSDK.maxFee);
+			expect(timestamp).toEqual(partialTransactionFromSDK.transactionInfo.timestamp);
+			expect(transactionBody).toEqual(partialTransactionFromSDK.transactionBody);
+			expect(status).toEqual(transactionStatus.detail.code);
+			expect(confirm).toEqual(transactionStatus.message);
+		});
+	});
```

Comment:
you have test for confirmed and partial, should you add one for unconfirmed?

Reply:
sure 👍🏼